### PR TITLE
"Feature: Google OAuth Configuration for Backend Authentication"

### PR DESCRIPTION
**Pull request by shikharpa**
This pull request introduces Google OAuth configuration to enable users to authenticate and sign in to our Django project's backend using their Google accounts. With this enhancement, users gain the convenience and security of logging in with their existing Google credentials, streamlining the authentication process and improving user experience. By integrating Google OAuth, we expand authentication options for users while maintaining robust security measures, enhancing the accessibility and usability of our application.

## Summary

- [x] Tested (Must)
- [ ] Test Case added
- [x] Build Successful (Must)
- [x] Sufficient Code comments added (Must)
- [ ] Attached Screenshots / Videos 
- [ ] All Relevant Documents added

### CODE REVIEW

File: services/api/kalvi/api/urls/__init__.py [hunk 1]
```diff
@@ -32,5 +32,10 @@
         "reset-password/<str:uid>/<str:token>/",
         UserPasswordResetView.as_view(),
         name="reset-password-through-mail",
+    ),
+    path(
+        "google/",
```
Comment (yaju09) [inline services/api/kalvi/api/urls/__init__.py:37]:
```suggestion
        "google/auth/",
```

File: services/api/db/Oauth/register.py [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+            raise AuthenticationFailed(
+                detail='Please continue your login using ' + filtered_user_by_email[0].auth_provider)
+    else:
+        #TODO: We need to provide a unique secret password for each social authentication user.
+        user = {
+            'name': generate_username(name), 'email': email,
+            'password': "SECRET" , 'confirm_password' : "SECRET"}
+        user = User.objects.create_user(**user)
+        if not user:
+            raise Exception ("We couldn't register user due to intenal issues! Please try after some time")
```
Comment (yaju09) [inline services/api/db/Oauth/register.py]:
```suggestion
            raise Exception ("We couldn't register user due to internal issues! Please try after some time")
```

**Review by yaju09 — CHANGES_REQUESTED** [review]

File: services/api/db/Oauth/register.py (hunk 2 above)
Comment (shikharpa) [reply services/api/db/Oauth/register.py]: changed it.

File: services/api/kalvi/api/urls/__init__.py (hunk 1 above)
Comment (shikharpa) [reply services/api/kalvi/api/urls/__init__.py:37]: this we will do in next chore work. when we will integrate with front end. (GitHub login URLs also same so i need to change both simultaneously )

**Comment by yogai53** [comment]
@shikharpa Please update your branch

**Pull request closed without merging**